### ISSUE TITLE
`cargo new --bin` fails with default VCS

### ISSUE THREAD

**Issue by azdavis** [rust-lang/cargo#4173]
```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.12.5
BuildVersion:	16F73
$ rustc --version
rustc 1.18.0
$ cargo --version
cargo 0.19.0-dev (28d1d60 2017-05-16)
$ git --version
git version 2.13.1
$ find foo
find: foo: No such file or directory
$ cargo new --bin --verbose foo
error: Failed to create project `foo` at `/Users/azdavis/foo`

Caused by:
  [2/-3] could not find '' to stat: No such file or directory
$ find foo
foo
foo/.git
$ cargo new --bin --vcs none bar
     Created binary (application) `bar` project
```

Is there something I'm missing?

Thanks for your help.

**Comment by alexcrichton** [comment]
Oh dear, sounds bad! Do you have a global `~/.gitconfig` perhaps? Sometimes libgit2 tries to interpret that or errors there manifest in errors in Cargo.

**Comment by azdavis** [comment]
I do have a global gitconfig (I actually keep mine at `~/.config/git/config`).

The problem seems to be this: in my global gitconfig, I have `init.templateDir = <the empty string>`. This is because I didn't want git populating new git repositories with things like `.git/description`, `.git/hooks/*.sample`, and `.git/info/exclude`, none of which I need by default. And setting `init.templateDir` to the empty string seemed to have the same effect as making `init.templateDir = <some path to a completely empty directory>`, which is what I wanted.

This also could explain the error from Cargo about "could not find '' to stat."

I did a quick perusal of the git man pages to see if setting `init.templateDir` to the empty string is an officially supported way to make it so that you just get an empty `init.templateDir`, but I couldn't find anything.

**Comment by alexcrichton** [comment]
Ah ok, thanks for the clarification! We've had [other bugs related to this](https://github.com/rust-lang/cargo/issues/1140) in the past but this may just be a part of how libgit2 handles the `templateDir` option? The previous issue [ended up getting fixed](https://github.com/libgit2/libgit2/issues/2814) in libgit2, want to open a bug over there for this?

**Comment by azdavis** [comment]
Sure: libgit2/libgit2#4271

**Comment by carols10cents** [comment]
update: [there is a fix in libgit2 for this](https://github.com/libgit2/libgit2/pull/4273), but it doesn't look to me like libgit2 has done a release since then for libgit2-sys to upgrade to yet (libgit2-sys is [currently at libgit2 15e1193, 0.26.0](https://github.com/alexcrichton/git2-rs/tree/master/libgit2-sys).

**Comment by dwijnand** [comment]
I think this is fixed (in master) now:

* the fix in libgit2 was a part of libgit2's v0.27.0-rc1 release,
* git2 was upgraded to libgit2 0.27.0-rc2 in alexcrichton/git2-rs#303, which was a part of git2's 0.7.0 release,
* cargo upgrade to git2 0.7.0 in #5091

**Comment by alexcrichton** [comment]
Awesome, thanks @dwijnand!
